pkg/gofr/datasource/scylladb/scylladb.go

```diff
@@ -0,0 +1,352 @@
+	sess, err := c.scylla.clusterConfig.createSession()
+
+	if err != nil {
+		c.logger.Error("failed to connect to ScyllaDB:", err)
+		return
+	}
+
+	scyllaBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
+	c.metrics.NewHistogram("app_scylla_stats", "Response time of ScyllaDB queries in microseconds", scyllaBuckets...)
+
+	c.logger.Logf("connected to '%s' keyspace at host '%s' and port '%d'", c.config.Keyspace, c.config.Host, c.config.Port)
+	c.scylla.session = sess
+}
+
+// UseLogger sets the logger for the scylladb client.
+func (c *Client) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the scylladb client.
+func (c *Client) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// UseTracer sets the tracer for the scylladb client.
+func (c *Client) UseTracer(tracer any) {
+	if tracer, ok := tracer.(trace.Tracer); ok {
+		c.tracer = tracer
+	}
+}
+
+// Query is the original method without context.
+// It internally delegates to QueryWithCtx using context.Background() as the default context.
+func (c *Client) Query(dest any, stmt string, values ...any) error {
+	return c.QueryWithCtx(context.Background(), dest, stmt, values...)
+}
+
+// Exec executes a CQL (Cassandra Query Language) statement on a ScyllaDB cluster
+// with the provided values, using the default context (context.Background()).
+func (c *Client) Exec(stmt string, values ...any) error {
+	return c.ExecWithCtx(context.Background(), stmt, values)
+}
+
+// ExecWithCtx executes a CQL statement by using the context,statement,values and returns error.
+func (c *Client) ExecWithCtx(ctx context.Context, stmt string, values ...any) error {
+	span := c.addTrace(ctx, "exec", stmt)
+	defer c.sendOperationStats(&QueryLog{Operation: "ExecWithCtx", Query: stmt, Keyspace: c.config.Keyspace}, time.Now(), "exec", span)
+
+	return c.scylla.session.Query(stmt, values...).Exec()
+}
+
+// ExecCAS performs Compare and Set operation on ScyllaDB cluster.
+func (c *Client) ExecCAS(dest any, stmt string, values ...any) (bool, error) {
+	return c.ExecCASWithCtx(context.Background(), dest, stmt, values)
+}
+
+// ExecCASWithCtx takes default context,destination,statement,values and  return bool and error.
+//
+//nolint:exhaustive // We just want to take care of slice and struct in this case.
+func (c *Client) ExecCASWithCtx(ctx context.Context, dest any, stmt string, values ...any) (bool, error) {
+	var (
+		applied bool
+		err     error
+	)
+
+	span := c.addTrace(ctx, "exec-cas", stmt)
+
+	defer c.sendOperationStats(&QueryLog{Operation: "ExecCASWithCtx", Query: stmt, Keyspace: c.config.Keyspace}, time.Now(), "exec-cas", span)
+
+	rvo := reflect.ValueOf(dest)
+	if rvo.Kind() != reflect.Ptr {
+		c.logger.Debugf("we did not get a pointer. data is not settable.")
```

Comment:
Here you are using a Debug

But here an error

https://github.com/gofr-dev/gofr/pull/1360#discussion_r1906554247